python/lsst/ts/standardscripts/auxtel/latiss_take_sequence.py

```diff
@@ -0,0 +1,248 @@
+                default: True
+            required: ["program", "reason"]
+        """
+        return yaml.safe_load(schema_yaml)
+
+    async def configure(self, config):
+        """Configure script.
+
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+            Script configuration, as defined by `schema`.
+        """
+
+        # make a list of tuples from the filter, exptime and grating lists
+        _recurrences = (
+            len(config.exposure_time_sequence)
+            if isinstance(config.exposure_time_sequence, collections.abc.Iterable)
+            else 1
+        )
+
+        self.visit_configs = [
+            (f, e, g)
+            for f, e, g in zip(
+                format_as_list(config.filter_sequence, _recurrences),
+                format_as_list(config.exposure_time_sequence, _recurrences),
+                format_as_list(config.grating_sequence, _recurrences),
+            )
+        ]
+
+        self.reason = config.reason
+
+        self.program = config.program
+
+        self.do_check_ataos_corrections = config.do_check_ataos_corrections
+
+    def set_metadata(self, metadata):
+        metadata.duration = 0
+        filters, gratings, exptime_total = set(), set(), 0
+        for filt, exptime, grating in self.visit_configs:
+            exptime_total += exptime
+            filters.add(filt)
+            gratings.add(grating)
+            metadata.duration += 3  # time to reconfigure latiss
+        metadata.duration += exptime_total
+        metadata.filter = f"{filters},{gratings}"
+
+    async def take_sequence(self):
+        """Take data while making sure ATCS is tracking."""
+
+        tasks = [
+            asyncio.create_task(self._take_sequence()),
+            asyncio.create_task(self.atcs.check_tracking()),
+        ]
+
+        await self.atcs.process_as_completed(tasks)
+
+    async def _take_sequence(self):
+        """Take the sequence of images as defined in visit_configs."""
+
+        nexp = len(self.visit_configs)
+
+        for i, (filt, exptime, grating) in enumerate(self.visit_configs):
+            await self.latiss.take_object(
+                exptime=exptime,
+                n=1,
+                filter=filt,
+                grating=grating,
+                group_id=self.group_id,
+                reason=self.reason,
+                program=self.program,
+            )
+
+            self.log.info(
+                f"Completed exposure {i + 1} of {nexp}. Exptime = {exptime:6.1f}s,"
+                f" filter={filt}, grating={grating})"
+            )
+
+    async def assert_feasibility(self) -> None:
+        """Verify that the telescope and camera are in a feasible state to
+        execute the script.
+        """
+
+        await self.latiss.assert_all_enabled()
+        await self.atcs.assert_all_enabled()
+
+        if self.do_check_ataos_corrections:
+            self.log.debug("Check ATAOS corrections are enabled.")
```

Comment:
I made a comment on the other PR about this.. we should move this to the `ATCS` class.. if you don't want to do it now, please, add a TODO and file a ticket.

Reply:
TODO added and ticket generated https://jira.lsstcorp.org/browse/DM-38823